src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -580,7 +580,9 @@ public function firstOrCreate(array $attributes = [], array $values = [])
     public function createOrFirst(array $attributes = [], array $values = [])
     {
         try {
-            return $this->create(array_merge($attributes, $values));
+            return $this->withSavepointIfNeeded(function () use ($attributes, $values) {
```

Comment:
I really liked how it read before: `createOrFirst` was essentially calling `create()` then `first()`, now it has more noise. I thought about pushing the savepoint to the `create()` method, which would keep this part as it was, but than it would be a bigger change.